infra-geo-ouverte/igo2-lib
# Style refactoring

This is a big one, we refactored all the library style. 
- No more magic import each file know what to do. 
- Remove scss-bundle
- Refactor the pipeline and migrate gulp script to use the native ng-packagr capabilities or node
- Allow style composition
- Remove gulp fix for @angular/cdk styles

**TO TEST THIS PR, YOU NEED TO:**
1. Check this branch
2. Delete the node_modules folder or @angular and stylus folder (unwanted fix is apply to @angular and stylus library)
3. Launch a clean install "npm ci"

**What is the current behavior?** (You can also link to an open issue here)
All the prebuilt themes was bundled and loaded in each app


**What is the new behavior?**
Compile prebuilt themes seperately and load only the theme we need.


**Does this PR introduce a breaking change?** (check one with "x")
- [x ] Yes

**If this PR contains a breaking change, please describe the impact and migration path for existing applications:**
The theming configuration has change. We need to check each application, who are not using prebuilt-themes. If your application configure manually the theme, please see the file demo/src/style.scss in the Demo and then the theme config

**Other information**:
